mozjs/js/rust/magic_codegen/src/lib.rs

```diff
@@ -321,13 +355,7 @@ fn gen_js_setter(id: &Option<syn::Ident>,
                         return false;
                     },
                 };
-                let v = match ToNumber(cx, call_args.index(0)) {
-                    Ok(num) => num,
-                    Err(_) => {
-                        JS_ReportErrorASCII(cx, b"Can't recognize arg 0\0".as_ptr() as *const libc::c_char);
-                        return false;
-                    },
-                };
+                get_js_val!(v, cx, call_args, 0);
```

Comment:
The way `get_js_val!` is now written, it can panic on invalid arguments, which is not an appropriate response to anything that can be called directly by JS. Instead it should set a pending exception on the `JSContext` and return false.